### PR TITLE
Add dynamic scheme/options sample

### PR DESCRIPTION
**Pull request by HaoK**
Pretty basic dynamic sample.

- Custom SimpleAuthHandler that uses SimpleOptions which just has a Displaymessage
- Home page lists the schemes (and remove them)
- Form lets you add schemes with a custom name and message

@Tratcher

### CODE REVIEW

File: samples/DynamicSchemes/Pages/Index.cshtml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+<div class="row">
+    <h2>Current Authentication Schemes</h2>
+    <ul>
+        @{
+            var schemes = await SchemeProvider.GetAllSchemesAsync();
+            foreach (var scheme in schemes)
+            {
+                <li><b>Scheme:</b> @scheme.Name -- <b>Handler:</b> @scheme.HandlerType.Name <b>DisplayMessage:</b> @Options.Get(scheme.Name).DisplayMessage <a asp-area="" asp-controller="Auth" asp-action="Remove" asp-route-scheme="@scheme.Name">Remove</a></li>
```
Comment (Tratcher) [inline samples/DynamicSchemes/Pages/Index.cshtml]: wrap?

**Review by Tratcher — COMMENTED** [review]
Screen shot?
The other related sample folks have asked for is updating options in place, like changing a ClientID. Add an Update endpoint that lets you change the message?

**Comment by HaoK** [comment]
I can just modify the Add form to allow update, since right now it throws on dupe, it can just become an update operation which gets rid of the ugly YSOD too

**Comment by Tratcher** [comment]
Sure, if you explain that on the form.

**Comment by HaoK** [comment]
a3df6a7bf9fc3f3dcd1164c0dcd90af21e3820f1

**Pull request closed without merging**